File: backend/app/schemas/audit.py
"""Audit log schemas."""

from pydantic import BaseModel, Field
from typing import Optional, List
from datetime import datetime
from uuid import UUID


class RequestResponse(BaseModel):
    """Request audit log response."""

    id: UUID
    trace_id: str
    org_id: UUID
    app_id: UUID
    env_id: Optional[UUID]
    user_id: Optional[str]
    session_id: Optional[str]
    scope: str
    input_preview: Optional[str]
    model: Optional[str]
    token_count: Optional[int]
    created_at: datetime

    class Config:
        from_attributes = True


class DecisionResponse(BaseModel):
    """Decision response."""

    id: UUID
    request_id: UUID
    trace_id: str
    policy_id: Optional[UUID]
    action: str
    reason: Optional[str]
    confidence: Optional[float]
    latency_ms: Optional[int]
    created_at: datetime

    class Config:
        from_attributes = True


class ViolationResponse(BaseModel):
    """Violation response."""

    id: UUID
    request_id: UUID
    trace_id: str
    check_name: str
    check_version: Optional[str]
    severity: str
    category: Optional[str]
    message: Optional[str]
    evidence: dict
    location_start: Optional[int]
    location_end: Optional[int]
    created_at: datetime

    class Config:
        from_attributes = True


class RequestDetailResponse(BaseModel):
    """Detailed request response with decisions and violations."""

    request: RequestResponse
    decisions: List[DecisionResponse]
    violations: List[ViolationResponse]


class AuditQueryParams(BaseModel):
    """Query parameters for audit log search."""

    app_id: Optional[UUID] = None
    user_id: Optional[str] = None
    session_id: Optional[str] = None
    start_date: Optional[datetime] = None
    end_date: Optional[datetime] = None
    scope: Optional[str] = None
    skip: int = Field(0, ge=0)
    limit: int = Field(100, ge=1, le=1000)

File: frontend/guardrails_frontend/src/types/audit.ts
// Audit Types (matching backend/app/schemas/audit.py)

export interface RequestResponse {
  id: string
  trace_id: string
  org_id: string
  app_id: string
  env_id?: string
  user_id?: string
  session_id?: string
  scope: string
  input_preview?: string
  model?: string
  token_count?: number
  created_at: string
}

export interface DecisionResponse {
  id: string
  request_id: string
  trace_id: string
  policy_id?: string
  action: string
  reason?: string
  confidence?: number
  latency_ms?: number
  created_at: string
}

export interface ViolationResponse {
  id: string
  request_id: string
  trace_id: string
  check_name: string
  check_version?: string
  severity: string
  category?: string
  message?: string
  evidence: Record<string, any>
  location_start?: number
  location_end?: number
  created_at: string
}

export interface RequestDetailResponse {
  request: RequestResponse
  decisions: DecisionResponse[]
  violations: ViolationResponse[]
}

